src/openms/source/ML/NNLS/NonNegativeLeastSquaresSolver.cpp

```diff
@@ -51,11 +100,11 @@ namespace OpenMS
 #endif
 
     // prepare solution array (directly copied from example)
-    double * x_vec = new double[a_cols + 1];
+    double * x_vec = new double[a_cols];
     double rnorm;
-    double * w = new double[a_cols + 1];
-    double * zz = new double[a_rows + 1];
-    int * indx = new int[a_cols + 1];
+    double * w = new double[a_cols];
+    double * zz = new double[a_rows];
+    int * indx = new int[a_cols];
     int mode;
```

Comment:
are you sure about this? This may segfault now

Reply:
probably fine